### PR TITLE
make panic customizable on each level

### PR DESCRIPTION
**Pull request by ysmood**
Related to #424

@egonelbre What do you think? If you think it's good, could you pick up this branch and work on it?

- [ ] implement browser helpers
- [ ] implement page helpers
- [ ] implement element helpers

### CODE REVIEW

File: browser.go [hunk 1]
```diff
@@ -270,6 +272,7 @@ func (b *Browser) PageFromTarget(targetID proto.TargetTargetID) (*Page, error) {
 	}
 
 	page = &Page{
+		e:         b.e,
```
Comment (ysmood) [inline browser.go:275]: Inherit parent's handler

File: lib/utils/utils.go [hunk 2]
```diff
@@ -60,11 +60,14 @@ func MultiLogger(list ...Logger) Log {
 	})
 }
 
+// Panic is the same as the built-in panic
+var Panic = func(v interface{}) { panic(v) }
```
Comment (ysmood) [inline lib/utils/utils.go:64]: Crazy user may override this global handler

**Comment by egonelbre** [comment]
Yeah, this would work as well. I'll finish it up.

Although, it would be nice to pass t.Fatal directly as argument, but it doesn't make a big difference for usage.

**Comment by ysmood** [comment]
> Although, it would be nice to pass t.Fatal directly as argument, but it doesn't make a big difference for usage.

Yes, I think it's better to make it more general, so people can customize it with more flexibility.

File: browser.go (hunk 1 above)
Comment (egonelbre) [reply browser.go:275]: Do you mean that this should be inherited into Element as well?

Comment (ysmood) [reply browser.go:275]: Yes, exactly! Recursive design.

Comment (ysmood) [reply browser.go:275]: This might help: https://github.com/go-rod/rod/blob/master/context.go

**Comment by egonelbre** [comment]
Btw. I'm not able to push to this branch?

Not sure whether this is intended, or should I make a new PR based on this?

**Comment by ysmood** [comment]
> Btw. I'm not able to push to this branch?
> 
> Not sure whether this is intended, or should I make a new PR based on this?

Yes, the reason is here:

image

**Comment by ysmood** [comment]
Close in favor of #433

**Pull request closed without merging**